routing/routing_helpers.cpp

```diff
@@ -18,6 +20,75 @@ namespace routing
 using namespace std;
 using namespace traffic;
 
+void FillSegmentInfo(vector<Segment> const & segments, vector<Junction> const & junctions,
+                     Route::TTurns const & turnDirs, Route::TStreets const & streets,
+                     Route::TTimes const & times, shared_ptr<TrafficStash> const & trafficStash,
+                     vector<Route::SegmentInfo> & segmentInfo)
+{
+  CHECK_EQUAL(segments.size() + 1, junctions.size(), ());
+  CHECK(!turnDirs.empty(), ());
+  CHECK(!times.empty(), ());
+
+  CHECK(std::is_sorted(times.cbegin(), times.cend(), my::LessBy(&Route::TTimeItem::first)), ());
+  CHECK(std::is_sorted(turnDirs.cbegin(), turnDirs.cend(), my::LessBy(&turns::TurnItem::m_index)), ());
+  CHECK(std::is_sorted(streets.cbegin(), streets.cend(), my::LessBy(&Route::TStreetItem::first)), ());
+
+  segmentInfo.clear();
+  if (segments.empty())
+    return;
+
+  segmentInfo.reserve(segments.size());
+  // Note. |turnDirs|, |streets| and |times| have point()|junctions| based index.
+  // On the other hand turns, street names and times are considered for further end of |segments| after conversion
+  // to |segmentInfo|. It means that street, turn and time information of zero point will be lost after
+  // conversion to |segmentInfo|.
+  size_t turnIdx = turnDirs[0].m_index != 0 ? 0 : 1;
+  size_t streetIdx = (!streets.empty() && streets[0].first != 0) != 0 ? 0 : 1;
+  size_t timeIdx = times[0].first != 0 ? 0 : 1;
+  double distFromBeginningMeters = 0.0;
+  double distFromBeginningMerc = 0.0;
+  double timeFromBeginningS = 0.0;
+
+  for (size_t i = 0; i < segments.size(); ++i)
+  {
+    size_t const segmentEndPointIdx = i + 1;
+
+    turns::TurnItem curTurn;
+    if (turnIdx != turnDirs.size() && turnDirs[turnIdx].m_index == segmentEndPointIdx)
+    {
+      curTurn = turnDirs[turnIdx];
+      if (turnIdx + 1 < turnDirs.size())
+        ++turnIdx;
+    }
+
+    string curStreet;
+    if (!streets.empty())
+    {
+      if (streetIdx != streets.size() && streets[streetIdx].first == segmentEndPointIdx)
+      {
+        curStreet = streets[streetIdx].second;
+        if (streetIdx + 1 < streets.size())
+          ++streetIdx;
+      }
+    }
+
+    if (timeIdx != times.size() && times[timeIdx].first <= segmentEndPointIdx)
+    {
+      timeFromBeginningS = times[timeIdx].second;
+      ++timeIdx;
+    }
+
+    distFromBeginningMeters +=
+      MercatorBounds::DistanceOnEarth(junctions[i].GetPoint(), junctions[i + 1].GetPoint());
+    distFromBeginningMerc += junctions[i].GetPoint().Length(junctions[i + 1].GetPoint());
```

Comment:
After the loop is finished it's a routeLengthMerc. But I need the value for every route segment. So as I see it's a distance from beginning. Don't you think so?